corehq/apps/fixtures/upload/run_upload.py

```diff
@@ -232,3 +200,46 @@ def clear_fixture_quickcache(domain, data_types):
     # the cache key that contains all types.
     global_type_ids = {dt.get_id for dt in FixtureDataType.by_domain(domain) if dt.is_global}
     get_fixture_items_for_data_types.clear(domain, global_type_ids)
+
+
+def _create_data_type(domain, table_def, replace, transaction):
+    errors = []
+
+    new_data_type = FixtureDataType(
+        domain=domain,
+        is_global=table_def.is_global,
+        tag=table_def.table_id,
+        fields=table_def.fields,
+        item_attributes=table_def.item_attributes
+    )
+    try:
+        existing_data_type = FixtureDataType.fixture_tag_exists(domain, new_data_type.tag)
+        if existing_data_type:
+            data_type = existing_data_type
+        # support old usage with 'UID'
+        elif table_def.uid:
+            data_type = FixtureDataType.get(table_def.uid)
+        else:
+            data_type = new_data_type
+
+        if replace and data_type != new_data_type:
+            data_type.recursive_delete(transaction)
+            data_type = new_data_type
+
+        data_type.fields = table_def.fields
+        data_type.item_attributes = table_def.item_attributes
+        data_type.is_global = table_def.is_global
+
+        if data_type.domain != domain:
+            data_type = new_data_type
+            errors.append(
+                _("'%(UID)s' is not a valid UID. But the new type is created.")
+                % {'UID': table_def.uid}
+            )
+        if table_def.delete:
+            data_type.recursive_delete(transaction)
+            return data_type, True
+    except (ResourceNotFound, KeyError):
```

Comment:
Is it possible to shrink the amount of code in the `try`?

Reply:
Probably, though I think that `ResourceNotFound` can be thrown at both the beginning and end (`recursive_delete`) of this. I'm not actually sure where `KeyError` could be thrown at the moment. There don't appear to be any dictionaries (would assume others would through `AttributeError`)